multiqc/modules/mosaicatcher/mosaicatcher.py

```diff
@@ -0,0 +1,182 @@
+import logging
+from collections import OrderedDict
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import bargraph, table
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="MosaiCatcher",
+            anchor="mosaicatcher",
+            href="https://github.com/friendsofstrandseq/mosaicatcher",
+            info=(
+                "Mosaicatcher counts Strand-seq reads and classifies strand states "
+                "of each chromosome in each cell using a Hidden Markov Model."
+            ),
+            doi="10.1038/s41587-019-0366-x",
+        )
+        log_files = self.find_log_files("mosaicatcher", filehandles=False)
+        if not log_files:
+            log.warning("No log files found for MosaiCatcher module")
+        else:
+            print("Log files found:", log_files)
+
+        samples = dict()
+        for f in list(log_files):
+            if not f or "f" not in f:
+                log.warning("Malformed log file object: {}".format(f))
+            else:
+                print("Parsing log file:", f["fn"])
+                samples = self._parse_samples(f, samples)
+                self.add_data_source(f)
+
+        self._samples = samples
+
+        self._add_table(samples)
+        self._add_coverage_plot(samples)
+
+        self.write_data_file(self._samples, "multiqc_mosaicatcher")
+        log.info("Found {} samples".format(len(samples)))
+
+    def _add_table(self, samples):
+        self.add_section(
+            plot=table.plot(data=samples, headers=self._setup_headers()),
+        )
+
+    def _add_coverage_plot(self, samples):
+        coverage_plot = bargraph.plot(
+            data=samples,
+            cats=["mapq", "dupl", "good", "reads2", "suppl"],
+            pconfig={
+                "id": "mosaicatcher-coverage",
+                "title": "MosaiCatcher: coverage",
+                "height": 1024,
+                "ymax": 4e6,
+                "ylab": "Cells",
+            },
+        )
+        self.add_section(
+            name="Average coverage per sample",
+            anchor="mosaicatcher-coverage-id",
+            description="Average coverage per sample",
```

Comment:
Would be good to use more descriptive category names (i.e. instead of `dupl`, use `Duplicate reads`, etc). And since we are summarising per cell, not per sample here, call the plot "Average coverage per cell" - same for other plots and metrics. And ideally add a more detailed description to the plot, i.e. how to interpret it in QC settings, if it makes sense

Reply:
Tried to be more descriptive here, done!